arrays/arrays.go

```diff
@@ -12,6 +12,19 @@ func FoldRight[T1, T2 any](as []T1, z T2, f func(T1, T2) T2) T2 {
 	return z
 }
 
+//The efficiency of this algorithm is O(N) and it does not reverse the list like FoldRight does.
+//TODO Make this function tail-recursive
```

Comment:
tail-recursion is a low-level runtime effect.  i don't think you can do Tail recursion in Go.  (Or in most languages with a C stack.  Go doesn't use a C stack, but I don't think it can do tail recursion.)